lib/puppet/defaults.rb

```diff
@@ -1570,7 +1570,11 @@ def self.default_diffargs
     :pluginsync => {
       :default    => true,
       :type       => :boolean,
-      :desc       => "Whether plugins should be synced with the central server.",
+      :desc       => "Whether plugins should be synced with the central server. This setting is deprecated,
+        as the decision to sync plugins with the central server is now based on whether a cached catalog is
+        being applied or not. If a cached catalog is being used, plugins will not be synced to ensure that
+        the same set of plugins are used that existed when the catalog was originally compiled.",
```

Comment:
I'd modify this to say `If a cached catalog is being used, this setting will be ignored and plugins will not be synced ...`.

Also, is there a standard width for these? Your description seems to be wider than most of the others.